Delete selected area

Luca-Ilari opened the pull request:
It's just a simple HotKey to delete the previously selected area 
Updated README with the HotKey 

https://user-images.githubusercontent.com/54724171/233442488-4e5e9e57-bb2b-4dfa-b407-3a679cf30acb.mp4

Hope it can be added and it's usefull :thumbsup:

Kully commented:
@Luca-Ilari Hello, and thank you so much for the PR! This looks pretty awesome. :)

I'm gonna check out the branch, play around and let you know how it feels to me. Will get back to you.

> Updated README with the HotKey

One of the things I was gonna ask for haha ... perhaps I should write a formal guideline for contributing? 🤔

Kully commented on README.md:
```diff
@@ -27,6 +27,7 @@ Make intuitve pixel art in this simple drawing tool written in _vanilla javascri
 | undo             | Z                   |
 | redo             | X                   |
 | color swap       | C                   |
+| delete selected area | Del                 | 
```
Can you move this line just above `remove selection`?

As this list is getting long, it would be best to group similar operations together.

Luca-Ilari replied:
I changed the line and added Backspace to the hotkeys column

Kully commented on js/script.js:
```diff
@@ -205,6 +205,30 @@ function Reset_Color_Of_Canvas_Cells()
+function Delete_Selected()
+{
+    const selection = document.getElementById("selection");
+    if( STATE["activeTool"] === "selection")
```
⚡️the blank space between `(` and `STATE` please

Kully commented on js/script.js:
```diff
@@ -205,6 +205,30 @@ function Reset_Color_Of_Canvas_Cells()
+    const selection = document.getElementById("selection");
+    if( STATE["activeTool"] === "selection")
+    {
+        if (selection != null)
```
Same ⚡️ comment here

Kully commented on js/script.js:
```diff
@@ -768,6 +792,10 @@ function Add_EventHandlers_To_Document()
             }
 
         }
+        if(e.code === "Delete")
```
I have a Mac, so I had to press `fn` then my `Delete` key to hit this if statement.

Can we add `Backspace` to this condition as well, so that users can always press one key?

```javascript
if(e.code === "Delete" || e.code === "Backspace")
{
    ...
}
```

@Luca-Ilari  WDYT? Any edge cases we're not considering?

Kully reviewed (changes requested):
Very nice work overall. Very clean commit, and love how you kept in sync with the CAMEL_CAPS I decided to go with when launching this (may switch to `thisCase` to keep standard, but idk 🤷🏻‍♂️)

Just a few comments and should be good to merge. 👍

Luca-Ilari commented:
Thanks for the review!
Changed line and added backspace hotkey to readme https://github.com/Kully/pixel-paint/pull/15/commits/649d9ea720de13d267c24d49e02b5498e54be3ab
Fixed the other problems https://github.com/Kully/pixel-paint/pull/15/commits/69ce664e788d2cba35cac0e548bb7b051091612d
I hope I didn't miss anything

Kully commented:
> Thanks for the review!
> Changed line and added backspace hotkey to readme [649d9ea](https://github.com/Kully/pixel-paint/commit/649d9ea720de13d267c24d49e02b5498e54be3ab)
> Fixed the other problems [69ce664](https://github.com/Kully/pixel-paint/commit/69ce664e788d2cba35cac0e548bb7b051091612d)
> I hope I didn't miss anything

Thank you! 🙏 Will give this a look and get back to you. :)

Kully commented:
@Luca-Ilari Also, out of curiosity, where did you find out about this project? 👀

Kully commented:
Looks good to me! 💃

Luca-Ilari commented:
> @Luca-Ilari Also, out of curiosity, where did you find out about this project? (Discord, Y Combinator, etc.) 👀

I was just cruising on the explore page on GitHub and found this cool project  :fire:

Kully commented:
Wow had no idea it made it up there!!

# Feels good

On Sat, Apr 22, 2023 at 7:20 PM Luca Ilari ***@***.***> wrote:

> @Luca-Ilari <https://github.com/Luca-Ilari> Also, out of curiosity, where
> did you find out about this project? (Discord, Y Combinator, etc.) 👀
>
> I was just cruising on the explore page on GitHub and found this cool
> project 🔥
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Kully/pixel-paint/pull/15#issuecomment-1518711174>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACPDQR57PEVCHXSZC4GAIT3XCQHPPANCNFSM6AAAAAAXF2SZMY>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
